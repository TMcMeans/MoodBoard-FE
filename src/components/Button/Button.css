.standard-btn {
  color: #ffffff;
  background: #DAA520;
  box-shadow:
    0px 4px 4px rgba(255, 255, 255, 0.25),
    inset 0px 10px 15px rgba(0, 0, 0, 0.25);
  border: none;
  border-radius: 5px;
  font-family: 'Raleway', sans-serif;
  font-weight: bold;
  font-size: 1.7rem;
  height: 3rem;
  width: 8rem;
}