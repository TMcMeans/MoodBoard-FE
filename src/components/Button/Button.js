import React from 'react';
import './Button.css';

export const Button = props => {
  const { text } = props;
  return <button className="standard-btn">{text}</button>;
};

export default Button;
