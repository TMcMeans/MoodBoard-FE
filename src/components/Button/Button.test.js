import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Button text="Sign in" />);
  });
  it('should render a snapshot with correct data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
