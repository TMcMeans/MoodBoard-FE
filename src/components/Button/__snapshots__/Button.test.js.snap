// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a snapshot with correct data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    text="Sign in"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Sign in",
      "className": "standard-btn",
    },
    "ref": null,
    "rendered": "Sign in",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Sign in",
        "className": "standard-btn",
      },
      "ref": null,
      "rendered": "Sign in",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
